pkg/openSearch/openSearchClient/client.go

```diff
@@ -99,21 +105,23 @@ func (c *Client) DeleteByQuery(indexName string, requestBody []byte) error {
 }
 
 func (c *Client) deleteByQuery(indexName string, requestBody []byte, isAsync bool) error {
-	deleteResponse, err := c.openSearchProjectClient.DeleteByQuery(
-		[]string{indexName},
-		bytes.NewReader(requestBody),
-		c.openSearchProjectClient.DeleteByQuery.WithWaitForCompletion(!isAsync),
+	waitForCompletion := !isAsync
+
+	_, err := c.openSearchProjectClient.Document.DeleteByQuery(
+		context.Background(),
+		opensearchapi.DocumentDeleteByQueryReq{
+			Indices: []string{indexName},
+			Body:    bytes.NewReader(requestBody),
+			Params: opensearchapi.DocumentDeleteByQueryParams{
+				WaitForCompletion: &waitForCompletion,
+			},
+		},
 	)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	resultString, err := io.ReadAll(deleteResponse.Body)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	return GetResponseError(deleteResponse.StatusCode, resultString, indexName)
+	return nil
```

Comment:
do we have to close the response body here?

Reply:
The status code and error is derived from the underlying HTTP request which means that the actual request body is not being read, so it does not have to be closed.